detectron2/utils/events.py

```diff
@@ -6,11 +6,19 @@
 import time
 from collections import defaultdict
 from contextlib import contextmanager
-from typing import Optional
+from typing import Dict, Optional, Union
 import torch
 from fvcore.common.history_buffer import HistoryBuffer
 
 from detectron2.utils.file_io import PathManager
+from detectron2.config import CfgNode
+
+try:
+    import wandb
+
+    assert hasattr(wandb, '__version__')  # verify package import not local dir
+except (ImportError, AssertionError):
+    wandb = None
```

Comment:
"Only import when needed" is how we deal with optional dependencies. So this better goes into the functions below (and there is no need to catch error then) .

Reply:
sounds good. will do

Reply:
this is done